### PR TITLE
Passing microseconds to the RoadRunner

### PR DESCRIPTION
**Pull request by msmakouz**
| Q             | A
| ------------- | ---
| Bugfix?       | ✔️
| Breaks BC?    | ❌
| New feature?  | ❌

See: https://github.com/roadrunner-server/lock/pull/10

### CODE REVIEW

**Review by butschster — APPROVED** [review]

File: src/Lock.php [hunk 1]
```diff
@@ -151,28 +151,28 @@ public function exists(string $resource, ?string $id = null): bool
...
-    private function convertTimeToSeconds(int|DateInterval $ttl): int
+    private function convertTimeToMicroseconds(int|float|DateInterval $ttl): int
     {
         if ($ttl instanceof DateInterval) {
-            return (int)$ttl->format('%s');
+            return (int) round((int)$ttl->format('%s') * 1000000);
         }
 
-        return $ttl;
+        return (int) round($ttl * 1000000);
```
Comment (butschster) [inline src/Lock.php]:
```suggestion
        return (int) round($ttl * 1_000_000);
```

File: src/Lock.php [hunk 2]
```diff
@@ -151,28 +151,28 @@ public function exists(string $resource, ?string $id = null): bool
...
 
         return $response->getOk();
     }
 
-    private function convertTimeToSeconds(int|DateInterval $ttl): int
+    private function convertTimeToMicroseconds(int|float|DateInterval $ttl): int
     {
         if ($ttl instanceof DateInterval) {
-            return (int)$ttl->format('%s');
+            return (int) round((int)$ttl->format('%s') * 1000000);
```
Comment (butschster) [inline src/Lock.php]:
```suggestion
            return (int) round((int)$ttl->format('%s') * 1_000_000);
```

File: tests/src/LockTest.php [hunk 3]
```diff
@@ -202,13 +202,15 @@ public static function updateTTLDataProvider(): \Traversable
...
     {
-        yield 'int' => [10, 10, 8, 8,];
+        yield 'int' => [10, 10000000, 8, 8000000,];
+
+        yield 'float' => [0.00001, 10, 0.000004, 4,];
 
         yield 'date-interval' => [
             new \DateInterval('PT10S'),
-            10,
+            10000000,
```
Comment (butschster) [inline tests/src/LockTest.php]:
```suggestion
            10_000_000,
```

File: tests/src/LockTest.php [hunk 4]
```diff
@@ -202,13 +202,15 @@ public static function updateTTLDataProvider(): \Traversable
...
             new \DateInterval('PT9S'),
-            9,
+            9000000,
```
Comment (butschster) [inline tests/src/LockTest.php]:
```suggestion
            9_000_000,
```

**Review by butschster — APPROVED** [review]

File: src/Lock.php (hunk 1 above)
Comment (msmakouz) [reply src/Lock.php]: Fixed, thanks.

File: src/Lock.php (hunk 2 above)
Comment (msmakouz) [reply src/Lock.php]: Fixed, thanks.

File: tests/src/LockTest.php (hunk 3 above)
Comment (msmakouz) [reply tests/src/LockTest.php]: Fixed, thanks.

File: tests/src/LockTest.php (hunk 4 above)
Comment (msmakouz) [reply tests/src/LockTest.php]: Fixed, thanks.

**Pull request merged**